encryptor/config/encryptionSettings.go

```diff
@@ -322,25 +358,58 @@ func (s *BasicColumnEncryptionSetting) Init() (err error) {
 	}
 
 	if s.DataType != "" {
-		dataType, err = common2.ParseStringEncryptedType(s.DataType)
+		if s.DataTypeID != 0 {
+			return common.ErrDataTypeWithDataTypeID
+		}
+
+		dataType, err = common.ParseStringEncryptedType(s.DataType)
 		if err != nil {
-			return fmt.Errorf("%s: %w", s.DataType, common2.ErrUnknownEncryptedType)
+			return fmt.Errorf("%s: %w", s.DataType, common.ErrUnknownEncryptedType)
 		}
-		if err = common2.ValidateEncryptedType(dataType); err != nil {
+		if err = common.ValidateEncryptedType(dataType); err != nil {
 			return err
 		}
 	}
 
+	if s.DataTypeID != 0 {
+		s.settingMask |= SettingDataTypeIDFlag
+
+		var dataTypeIDEncoders = type_awareness.GetPostgreSQLDataTypeIDEncoders()
+		if useMySQL {
+			//dataTypeIDEncoders = base.GetMySQLDataTypeIDEncoders()
```

Comment:
lets return error and fail validation due to unsupported on MySQL?

Reply:
Discussed leaving this PR unmerged until implementing corresponding logic for MySQL.